sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -713,9 +713,7 @@ class StreamSuite extends StreamTest {
         "columnName" -> "`rn_col`",
         "windowSpec" ->
           ("(PARTITION BY COL1 ORDER BY COL2 ASC NULLS FIRST ROWS BETWEEN UNBOUNDED PRECEDING " +
-          "AND CURRENT ROW)")),
-      queryContext = Array(
-        ExpectedContext(fragment = "withColumn", callSitePattern = getCurrentClassCallSitePattern)))
```

Comment:
How about to revert changes for the `withColumn()` method?

Reply:
`withColumn` takes `Column` as parameters, why do we lose the error context if `Column` captures it?

Reply:
Because when we throw the exception at `UnsupportedOperationChecker`:
<img width="847" alt="Screenshot 2024-01-10 at 16 45 27" src="https://github.com/apache/spark/assets/1580697/5358acbb-d5cb-46c5-9f8a-ff490c23fead">
`origin` of `Window` is empty. It contains nothing because when we construct it at `Analyzer`, we put captured column context inside of a sequence of `windowExpressions`:
<img width="835" alt="Screenshot 2024-01-10 at 16 58 00" src="https://github.com/apache/spark/assets/1580697/c44838da-e427-42cf-993d-f2ffd4c6fdff">

Reply:
I have added `origin` from the head expression:
```scala
          throw QueryExecutionErrors.nonTimeWindowNotSupportedInStreamingError(
            windowFuncList,
            columnNameList,
            windowSpecList,
            windowExpression.head.origin)
```
for double check. In that case, the query context is propagated obviously.

Reply:
Here is the related PR: https://github.com/apache/spark/pull/41578

Reply:
@cloud-fan So far, I would revert changes in `withColumn`. Need to figure out how to properly fix the case of `Window` + Dataframe query context.

Reply:
then we should add `withOrigin` in `select` as well? Any API that produces `Project` or `Window` should have error context.